Source/WebCore/platform/ScrollSnapAnimatorState.cpp

```diff
@@ -166,56 +182,54 @@ bool ScrollSnapAnimatorState::resnapAfterLayout(ScrollOffset scrollOffset, const
     auto snapOffsetsVertical = snapOffsetsForAxis(ScrollEventAxis::Vertical);
     auto snapOffsetsHorizontal = snapOffsetsForAxis(ScrollEventAxis::Horizontal);
     
+    auto previouslySnappedBoxes = std::exchange(m_currentlySnappedBoxes, { });
+    
     // Check if we need to set the current indices
     if (!activeVerticalIndex || *activeVerticalIndex >= snapOffsetsForAxis(ScrollEventAxis::Vertical).size()) {
-        snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffset(ScrollEventAxis::Vertical, scrollOffset, scrollExtents, pageScale);
+        snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffsetInternal(ScrollEventAxis::Vertical, scrollOffset, scrollExtents, pageScale);
         activeVerticalIndex = activeSnapIndexForAxis(ScrollEventAxis::Vertical);
     }
     if (!activeHorizontalIndex || *activeHorizontalIndex >= snapOffsetsForAxis(ScrollEventAxis::Horizontal).size()) {
-        snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffset(ScrollEventAxis::Horizontal, scrollOffset, scrollExtents, pageScale);
+        snapPointChanged |= setNearestScrollSnapIndexForAxisAndOffsetInternal(ScrollEventAxis::Horizontal, scrollOffset, scrollExtents, pageScale);
         activeHorizontalIndex = activeSnapIndexForAxis(ScrollEventAxis::Horizontal);
     }
-    
-    auto horizontalOffsets = currentlySnappedOffsetsForAxis(ScrollEventAxis::Horizontal);
-    auto verticalOffsets = currentlySnappedOffsetsForAxis(ScrollEventAxis::Vertical);
 
-    auto currentlySnapped = currentlySnappedBoxes(horizontalOffsets, verticalOffsets);
-    
-    auto wasSnappedToMultipleBoxes = m_currentlySnappedBoxes.size() > 1;
-    auto currentlySnappedToMultipleBoxes = currentlySnapped.size() > 1;
+    updateCurrentlySnappedBoxes();
```

Comment:
Rather than updating m_currentlySnappedBoxes here maybe make a helper function that uses currentlySnappedBoxes? Also, to make my above thought clearer, maybe currentlySnappedBoxes should be renamed to possibleBoxesToSnapTo or something similar.

Reply:
It does feel like `m_currentlySnappedBoxes` could probably use a better name; before testing it myself, I was under the impression that `m_currentlySnappedBoxes` tracked the current elements we've already snapped to, rather than the current box(es) we're targeting. Maybe something along the lines of `m_ possibleBoxesToSnapTo`, `m_candidateBoxesForSnapping` or `m_candidateBoxesToSnapTo`?